scripts/entity/components/input_components/base_input_component.gd

```diff
@@ -0,0 +1,65 @@
+## A base class that is meant to be overwrote in order to provide a conssitent
+## way to deal with input while also providing most basic logic.
+class_name BaseInputComponent extends Node
+
+## Event called when the jump input starts.
+signal on_jump_input_started()
+## Event called when the jump input is held.
+signal on_jump_input()
+## Event called when the jump input releases.
+signal on_jump_input_cancelled()
+
+## Event called when the horizontal input is held.
+## [br]
+## [param horizontal_value] float: The horizontal input value.
+signal on_horizontal_movement_input(horizontal_value: float)
+
+
+@export var JUMP_INPUT_NAME = "jump"
+@export var LEFT_INPUT_NAME = "left"
+@export var RIGHT_INPUT_NAME = "right"
```

Comment:
Why not const. Why in base rather than player specific

Reply:
Moved out to player. Now it is a const but must be changed in code rather than the editor